### ISSUE TITLE
No obvious way to map modifiers, no way to record, nearly unusable UI

### ISSUE THREAD

**Issue by gitcnd** [tekezo/Karabiner-Elements#1646]
Awesome product guys - but ruined by a hostile user interface and poor usage planning.

For example - I want to map Ctrl-Home on my PC keyboard, so it send the same as Command-Up-Arrow on my mac, and preferably only when I'm inside my Arduino IDE...

I was expecting a "Record new Key" option someplace, where from in that I would see this:

a) Press the key you want to map from
[so I hit Ctrl-Home]

b) Press the key you want to map Ctrl-Home to:
[so I hit Command-Uparrow]

and perhaps someplace in the options, a way to restrict this mapping so it only takes effect when I'm in selected apps - for example:-

c) You have 5 seconds to switch to the app you want this key mapping to be restricted to
[so I open Arduino, and it detects what I'm in and adds this to the mapping setup]

========

Aside from that:-
a) in the existing simple modifications UI, there's no way to assign modifiers (Ctrl) to keys - only ways to map entire keys.
b) in the complex modifications UI, there's no way to add new rules (only import, whatever that means)

Long story short - it looks like your code does what I want, except you UI does not have any way to let me reach that.

**Comment by clemsam** [comment]
_Huhu, nice to read a merry satire in such a dry-humoured place … **thanks for the cheering-up!**_

Well, what did you do when you expected a "Record new Key" option and there was none?!?
Obviously you condescended to write a letter to donate people your precious opinion.

Well, go on, pay some money for a commercial product and harass their support when you "expect something" that isn't right where "you expect it". (Those people have lots of experience to soothe any kind of weird people, with the highest courtesy.)

Now, just for the other users who happen onto this skirmish:  
As most of you have read in [Karabiner.json Reference Manual](https://pqrs.org/osx/karabiner/json.html) you can put any self-composed complex modifications into your own `Users/userName/.config/karabiner/assets/ complex_modifications` folder and import them from there into the GUI.

You will also have read that it's quite easy to:
 – Condition any remapping to any application
 – Remap any modifier-&-key combination to any other one
 – Open any application via osascript (Terminal or AppleScript)
 – Display modal windows asking for information via osascript-AppleScript

_Congratulations to all users who had the common sense to look for a Manual and were actually capable of reading AND understanding it!_

**Comment by gitcnd** [comment]
I invested considerable effort show you how to improve your product.  99.9% of others would not.
You should kick everyone who gives you helpful suggestion in the teeth.

Oh wait...

**Comment by gsjackson** [comment]
Chris you strike me as a tool.  Be useful, screw your self elsewhere.

**Comment by clemsam** [comment]
Btw: I searched and found that from 1,646 issues open or closed here, yours is the ONLY one with more than one "no(t)" in its title and the ONLY one that presumes to give a "final judgement" without having cared to ask this community for help. _"So saaad"_ … ;-)

_YOU must be a very lonely guy …_
